src/controllers/csv-processor.ts

```diff
@@ -101,16 +102,75 @@ function validateParams(page_number: number, max_page_number: number, page_size:
     return errors;
 }
 
-export const uploadCSV = async (buff: Buffer, dataset: Dataset): Promise<UploadDTO | UploadErrDTO> => {
+export const moveCSVFromBlobStorageToDatalake = async (dataset: Dataset): Promise<UploadDTO | UploadErrDTO> => {
+    const blobStorageService = new BlobStorageService();
     const dataLakeService = new DataLakeService();
+    const datafiles = await dataset.datafiles;
+    const datafile: Datafile | undefined = datafiles
+        .filter((filterfile: Datafile) => filterfile.draft === false)
+        .sort(
+            (first: Datafile, second: Datafile) =>
+                new Date(second.creationDate).getTime() - new Date(first.creationDate).getTime()
+        )
+        .shift();
+    const dto = await datasetToDatasetDTO(dataset);
+    if (datafile) {
+        try {
+            logger.debug(`Moving file ${datafile.id} from blob storage to datalake`);
+            const buff: Buffer = await blobStorageService.readFileToBuffer(`${datafile.id}.csv`);
+            await dataLakeService.uploadFile(`${datafile.id}.csv`, buff);
+            datafile.draft = false;
+            await datafile.save();
+            await blobStorageService.deleteFile(`${datafile.id}.csv`);
+            return {
+                success: true,
+                dataset: dto
+            };
+        } catch (err) {
+            logger.error(err);
+            return {
+                success: false,
+                dataset: dto,
+                errors: [
+                    {
+                        field: 'csv',
+                        message: [
+                            { lang: ENGLISH, message: t('errors.move_to_datalake', { lng: ENGLISH }) },
+                            { lang: WELSH, message: t('errors.move_to_datalake', { lng: WELSH }) }
+                        ],
+                        tag: { name: 'errors.move_to_datalake', params: {} }
+                    }
+                ]
+            };
+        }
+    } else {
```

Comment:
Its here because API requests are atomic... There is a possible state where there is no datafile attached to the Dataset.  This can occur because the user decided they were unhappy with the file they uploaded.  In this situation you end up in a position where there's a dataset but no datafile.  At this point a second user could make this request and there be no datafile to move.  So it can happen, it shouldn't happen but we have to code on the assumption that it will happen.